snappy/overlord.go

```diff
@@ -246,6 +263,57 @@ func ActivateSnap(s *Snap, inter interacter) error {
 	return os.Symlink(filepath.Base(s.basedir), currentDataSymlink)
 }
 
+func UndoUpdateCurrentSymlink(oldSnap, newSnap *Snap, inter interacter) error {
+	removeCurrentSymlink(newSnap, inter)
+	return UpdateCurrentSymlink(oldSnap, inter)
+}
+
+func removeCurrentSymlink(s *Snap, inter interacter) error {
+	currentActiveSymlink := filepath.Join(s.basedir, "..", "current")
+
+	// and finally the current symlink
+	if err := os.Remove(currentActiveSymlink); err != nil {
+		logger.Noticef("Failed to remove %q: %v", currentActiveSymlink, err)
+	}
+
+	currentDataSymlink := filepath.Join(dirs.SnapDataDir, s.Name(), "current")
+	if err := os.Remove(currentDataSymlink); err != nil && !os.IsNotExist(err) {
+		logger.Noticef("Failed to remove %q: %v", currentDataSymlink, err)
+	}
+
+	return nil
+}
+
+// ActivateSnap is a wrapper around
+// (generate-security-profile, generate-wrappers, update-current-symlink)
+func ActivateSnap(s *Snap, inter interacter) error {
+	currentActiveSymlink := filepath.Join(s.basedir, "..", "current")
+	currentActiveDir, _ := filepath.EvalSymlinks(currentActiveSymlink)
```

Comment:
The err result here shouldn't be ignored. Below we take the lack of a currentActiveDir as a sign the snap is not activated, but it might have been an error.